Samples/Starbucks-sample/Starbucks-sample/Overlay/Results/ResultsViewController.m

```diff
@@ -0,0 +1,236 @@
+    UIFont *labelValueFont = [UIFont systemFontOfSize:kValueLabelFontSize weight:UIFontWeightRegular];
+
+    UIColor *labelKeyColor = [UIColor MB_purpleyGreyColor];
+    UIColor *labelValueColor = [UIColor MB_slateGreyColor];
+
+    NSArray *keyArray = [self.labelMap allKeys];
+
+    CGFloat height = 0;
+
+    UILabel *lastLabel = nil;
+    for (int i = 0; i < keyArray.count; ++i) {
+        NSString *key = keyArray[i];
+        NSString *value = [self.labelMap objectForKey:key];
+
+        UILabel *labelValue = [self createLabelWithFont:labelValueFont andTextColor:labelValueColor andText:value];
+        [resultContainerView addSubview:labelValue];
+
+        [labelValue.leadingAnchor constraintEqualToAnchor:resultContainerView.leadingAnchor
+                                                 constant:kLeadingResultViewToLeadingLabelDistance]
+            .active = YES;
+        [labelValue.trailingAnchor constraintEqualToAnchor:resultContainerView.trailingAnchor
+                                                  constant:-kTrailingResultViewToTrailingLabelDistance]
+            .active = YES;
+
+        CGSize neededValueSize = [labelValue sizeThatFits:CGSizeMake(CGFLOAT_MAX, CGFLOAT_MAX)];
+
+        if (i == 0) {
+            [labelValue.bottomAnchor constraintEqualToAnchor:resultContainerView.bottomAnchor
+                                                    constant:-kBottomResultViewToBottomValueLabelDistance]
+                .active = YES;
+            height += neededValueSize.height + kBottomResultViewToBottomValueLabelDistance;
+        } else {
+            [labelValue.bottomAnchor constraintEqualToAnchor:lastLabel.topAnchor constant:-kLabelToLabelDistance].active = YES;
+            height += kLabelToLabelDistance + neededValueSize.height;
+        }
+
+        UILabel *labelKey = [self createLabelWithFont:labelKeyFont andTextColor:labelKeyColor andText:key];
+        [resultContainerView addSubview:labelKey];
+
+        [labelKey.bottomAnchor constraintEqualToAnchor:labelValue.topAnchor constant:-kLabelToLabelDistance].active = YES;
+        [labelKey.leadingAnchor constraintEqualToAnchor:resultContainerView.leadingAnchor constant:kLeadingResultViewToLeadingLabelDistance]
+            .active = YES;
+        [labelKey.trailingAnchor constraintEqualToAnchor:resultContainerView.trailingAnchor
+                                                constant:-kTrailingResultViewToTrailingLabelDistance]
+            .active = YES;
+
+        CGSize neededKeySize = [labelKey sizeThatFits:CGSizeMake(CGFLOAT_MAX, CGFLOAT_MAX)];
+        height += kLabelToLabelDistance + neededKeySize.height;
+
+        lastLabel = labelKey;
+    }
+
+    [lastLabel.topAnchor constraintEqualToAnchor:resultContainerView.topAnchor constant:kTopResultViewToTopValueLabelDistance].active = YES;
+    height += kTopResultViewToTopValueLabelDistance;
+
+    [resultContainerView.heightAnchor constraintEqualToConstant:height];
```

Comment:
Long methods such as this one should be structured into building blocks wherever possible.

- setupCloseButton:(UIButton *)button
- setupSubmitButton:(UIButton *)submitButton
- setupResultsContainerView:(UIView*)containerView

This provides more clarity which part of code is responsible for